### ISSUE TITLE
Lots of test failures with PyPy3.10 7.3.12

### ISSUE THREAD

**Issue by mgorny** [pytest-dev/pyfakefs#859]
**Describe the bug**
When running the test suite with PyPy3.10 7.3.12, I get loads of test failures. The following bit seems to repeat a lot:

```pytb
Traceback (most recent call last):
  File "/tmp/pyfakefs/pyfakefs/tests/fake_filesystem_test.py", line 1433, in test_alt_sep
    fake_os_module = fake_os.FakeOsModule(self.filesystem)
  File "/tmp/pyfakefs/pyfakefs/fake_os.py", line 1356, in wrapped
    return f(*args, **kwargs)
TypeError: object.__new__() takes exactly one argument (the type to instantiate)
```

Full log (1M): [pypy310.txt](https://github.com/pytest-dev/pyfakefs/files/11988850/pypy310.txt)

**How To Reproduce**
```
tox -e pypy310
```

**Your environment**
Please run the following in the environment where the problem happened and
paste the output.
```bash
$ python -c "import platform; print(platform.platform())"
Linux-6.3.11-gentoo-dist-x86_64-AMD_Ryzen_5_3600_6-Core_Processor-with-glibc2.37
$ python -c "import sys; print('Python', sys.version)"
Python 3.10.12 (af44d0b8114cb82c40a07bb9ee9c1ca8a1b3688c, Jun 16 2023, 13:56:29)
[PyPy 7.3.12 with GCC 12.2.1 20230428]
$ python -c "from pyfakefs import __version__; print('pyfakefs', __version__)"
pyfakefs 5.3.dev0
$ python -c "import pytest; print('pytest', pytest.__version__)"
pytest 7.4.0
```

I've originally noticed it on 5.2.2, and tested on 78d173b4c111e1394f906d667e45775afcb8fe65 to confirm.

**Comment by mrbean-bremen** [comment]
Thank you, I will have a look!

**Comment by mrbean-bremen** [comment]
@mgorny - shall be fixed in main now, please check!

**Comment by mgorny** [comment]
Thanks. I can confirm that it passes now, except for some assertion from pandas — but that's not a bug in pyfakefs.

**Comment by mrbean-bremen** [comment]
Do you need a patch release for packaging, or can this wait?

**Comment by mgorny** [comment]
I would find a new release very helpful but if that's a lot of trouble, we can try to backport the patches in Gentoo.

**Comment by mrbean-bremen** [comment]
Sure, a release is no touble at all. Released as [5.2.3](https://pypi.org/project/pyfakefs/5.2.3/), and closing as fixed.

**Comment by mgorny** [comment]
Thanks again!
